src/utils/list_cpu_features.c

```diff
@@ -30,7 +36,176 @@
 #include "cpuinfo_ppc.h"
 #endif
 
-static void PrintEscapedAscii(const char* str) {
+// Design principles
+// -----------------
+// We build a tree structure containing all the data to be displayed.
+// Then depending on the output type (text or json) we walk the tree and display
+// the data accordingly.
+
+// We use a bump allocator to allocate strings and nodes of the tree,
+// Memory is not intented to be reclaimed.
+typedef struct {
+  char* ptr;
+  size_t size;
+} BumpAllocator;
+
+// Allocate a buffer of size `size`.
+static BumpAllocator BA_Create(size_t size) {
+  char* const ptr = (char*)malloc(size);
```

Comment:
For something that simple (i.e. a 64 KiB buffer used by a single program), I would recommend just using a global static buffer and BumpAllocator variable. This would ensure everything is clear on startup (see comments below), and would simply all functions to not need to |BA| parameter.